README


This archive contains the distribution JSON-Any,
version 1.32:

  Wrapper Class for the various JSON classes.

This software is copyright (c) 2013 by Chris Thompson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




CHANGES
==================================================
Changes from 2012-11-07 00:00:00 +0000 to present.
==================================================

-----------------------------------------
version 1.32 at 2013-11-07 15:56:33 +0000
-----------------------------------------

  Change: e1fbae5ccddc7becce7b1bcff1a8ac99075bc136
  Author: Chris Prather <chris@prather.org>
  Date : 2013-11-07 07:52:41 +0000

    Merge pull request #3 from wolfsage/t_fix_deps

    Include dep Test::Requires in dist.ini 

  Change: 0880faaaf3c8194acc3a70a387d94ff69633b333
  Author: Matthew Horsfall <wolfsage@gmail.com>
  Date : 2013-11-07 10:51:02 +0000

    Include dep Test::Requires in dist.ini 

-----------------------------------------
version 1.31 at 2013-11-06 21:46:52 +0000
-----------------------------------------

  Change: 87ce2b21a27c984d71ec9b2a94847ae99d48445b
  Author: Chris Prather <chris@prather.org>
  Date : 2013-11-06 21:46:52 +0000

    ignore a few more directories 

  Change: d585c4cdccb59c96d7a3ab78e1881edd7e4d4e59
  Author: Chris Prather <chris@prather.org>
  Date : 2013-11-06 21:44:04 +0000

    bump minimum version requirement for JSON.pm 

  Change: f5200618645fc09d27856b921bc9be43cf35b039
  Author: Chris Prather <chris@prather.org>
  Date : 2013-11-06 21:41:22 +0000

    fix the tests to properly exclude JSON::XS

    A bug came up where JSON::XS and JSON.pm had incompatible revisions
    but installed just fine. This was a problem because JSON.pm will
    silently load JSON::XS in the background and caused explosions.

    We now try to exclude JSON::XS except in the case we're actually
    testing it directly.

    Also we've removed some of the tests for JSON::Syck as it's been
    deprecated. We also add in tests for Cpanel::JSON::XS which got
    missed when it was added as a choice. 

  Change: 14744d6d0cce26bc867638ec5aaea9594c292efe
  Author: Chris Prather <chris@prather.org>
  Date : 2013-10-31 22:45:19 +0000

    add a todo note for ether (or whomever) to pick up before I get back
    to it 

  Change: 360853b664817a0ed0e4b473dfbf1dbf1e56dd77
  Author: Chris Prather <chris@prather.org>
  Date : 2013-10-31 17:51:41 +0000

    test without JSON::XS and add a compat note

    JSON::XS 3.x is incompatible with JSON.pm < 2.90 in the exact way
    that JSON::Any uses things in the test suite.

    So we set the tests to ignore JSON::XS (using Test::Without::Module)
    thanks to ether for suggesting it.

    And we add a note about the situation to the POD so that others can
    figure out what is going on. 

  Change: 756c6ac460c92d425d970e113465ce1e9a6b40a9
  Author: Chris Prather <chris@prather.org>
  Date : 2013-10-31 15:29:16 +0000

    test for more accpetable backends on install

    We have added more acceptable backends (Cpanel::JSON::XS and
    JSON::PP) but we didn't update the tests during installation. 

  Change: cc51c101a1aa0d45573d25cd6bad3c9eb07006f5
  Author: Chris Prather <chris@prather.org>
  Date : 2013-10-31 14:59:45 +0000

    Pull Changes from Git and prune local/

    Migrating out Changes file to ChangesFromGit so I won't have to
    remember to update it manually. If people desperately need the old
    commit messages broken out by release I can go back and tag the
    releases based on the old changes file.

    Additional to prevent myself from pulling a "perigrin" as rjbs put
    it, I've added a directive to prune Carton stile local/ directories. 

  Change: ed357ecabed9ede1cfffb021f5c922a4bcc5c828
  Author: Chris Prather <chris@prather.org>
  Date : 2013-10-30 06:19:58 +0000

    Merge pull request #2 from ilmari/json-xs-3

    Add support for JSON::XS version 3 

  Change: 6bd1def2e2fddfd79aff4b36e2f05519bc89366e
  Author: Dagfinn Ilmari Mannsåker <ilmari@ilmari.org>
  Date : 2013-10-30 11:19:24 +0000

    Add support for JSON::XS version 3

    For added future robustness, skip modules we manage to load but don't
    know how to handle. 

  Change: 75a29b8203bc430481b5afed2a08a78e52edc037
  Author: Dagfinn Ilmari Mannsåker <ilmari@ilmari.org>
  Date : 2013-10-30 11:15:55 +0000

    Test string escaping with JSON:PP as well 

  Change: f8fb91cf5911ea5022883a34d69f362bbf5f59a4
  Author: Dagfinn Ilmari Mannsåker <ilmari@ilmari.org>
  Date : 2013-10-30 11:15:55 +0000

    Fix no_plan usage 

-----------------------------------------
version 1.30 at 2013-06-15 20:42:55 +0000
-----------------------------------------

  Change: 9855652fcb7cfcb6c66a846d0cb6b124004239d3
  Author: Chris Prather <chris@prather.org>
  Date : 2013-06-15 16:42:55 +0000

    clean up and modernize the dist for 1.30 release 

  Change: d324166b5e6b38625bc13e5cba4116365305dad5
  Author: Chris Prather <chris@prather.org>
  Date : 2013-06-15 16:17:02 +0000

    tidy up the code 

  Change: b20a5d160371ebec9c90f36a942b79fabd44aaf2
  Author: Chris Prather <chris@prather.org>
  Date : 2013-06-15 16:16:11 +0000

    add Cpanel::JSON::XS support 

  Change: 84157525e8684ed812e7007484ad208da2630f48
  Author: Chris Prather <chris@prather.org>
  Date : 2013-06-15 16:14:59 +0000

    add support for JSON::PP 

  Change: edbbf5fac3effd77e89908c37bd569f6760fc3f4
  Author: Chris Prather <chris@prather.org>
  Date : 2011-09-17 20:29:55 +0000

    have dzil manage our version numbers 

  Change: 34965f9f6daecad7b57c4e2898cdc34a436d3674
  Author: Chris Prather <chris@prather.org>
  Date : 2011-09-17 20:19:57 +0000

    rework to use a similar process as Any::Moose so that the toolchain
    bits work properly 

  Change: 46df74c03caed6f8226ddf6953f476f88831b753
  Author: Chris Prather <chris@prather.org>
  Date : 2011-09-17 20:18:49 +0000

    update version and changes for release 

  Change: 90acf0e8e34b949db9d0af1a3655e833efcc7e81
  Author: Chris Prather <chris@prather.org>
  Date : 2011-06-07 19:58:30 +0000

    fix for version number collision 

  Change: c4fc946bf27599fd3c4312a93d6485df0ea4102d
  Author: Chris Prather <chris@prather.org>
  Date : 2011-06-07 19:54:26 +0000


Makefile.PL
# This Makefile.PL for JSON-Any was generated by
# inc::MakeMaker <self>.
# Don't edit it but the dist.ini used to construct it.

use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Wrapper Class for the various JSON classes.",
  "AUTHOR" => "Chris Thompson <cthom\@cpan.org>, Chris Prather <chris\@prather.org>, Robin Berjon <robin\@berjon.com>, Marc Mims <marc\@questright.com>, Tomas Doran <bobtfish\@bobtfish.net>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "JSON-Any",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "JSON::Any",
  "PREREQ_PM" => {},
  "TEST_REQUIRES" => {
    "Test::Requires" => 0,
    "Test::Without::Module" => "0.17"
  },
  "VERSION" => "1.32",
  "test" => {
    "TESTS" => "t/*.t"
  }
);
;

  
    sub has_json () {
        our @order = qw(CPANEL XS PP JSON DWIW);
        foreach my $testmod (@order) {
            $testmod = "JSON::$testmod" unless $testmod eq "JSON";
            $testmod = "Cpanel::JSON::XS" if $testmod eq "JSON::CPANEL";
            eval "require $testmod";
            return 1 unless $@;
        }
        return 0;
    }

    if (has_json) {
        # we have some kind of supported JSON module, we're good
    }
    else {

        # we need to have a version of JSON, go with JSON.pm as a sane default
        $WriteMakefileArgs{PREREQ_PM}{JSON} = '2.90';
    }
    
  


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



